docs/spec_draft.md

```diff
@@ -430,6 +431,66 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.bitcast_convert
+
+### Semantics
+
+Performs an element-wise bitcast operation on `operand` tensor and produces a
+`result` tensor where the bits are reinterpreted with the new element type.
+Conversions between different bitwidths are not element-wise and creates/deletes
+a dimension if new element type requires less/more bits, respectively.
+
+Let `E` and `E'` be the `operand` and `result` element type respectively, and
+`R = rank(operand)`:
+  * If `numBits(E)` $=$ `numBits(E')`,
+     `bitRepr(result[i0, ..., iR-1]) = bitRepr(operand[i0, ..., iR-1])`.
+  * If `numBits(E)` $\gt$ `numBits(E')`,
+    `bitRepr(result[i0, ..., iR-1, :]) = bitRepr(operand[i0, ..., iR-1])`.
+  * If `numBits(E)` $\lt$ `numBits(E')`,
+    `bitRepr(result[i0, ..., iR-2]) = bitRepr(operand[i0, ..., iR-2, :])`.
+
+Bitcast is implemented as a low-level cast, so machines with different
+floating-point representations (e.g. endianesss) will give different results.
+
+### Inputs
+
+| Name      | Type                         |
+|-----------|------------------------------|
+| `operand` | tensor of any supported type |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same shape except for the last
+    dimension. Let `E` and `E'` be the `operand` and `result` element type,
+    respectively and `R = rank(operand)`:
+    * If `numBits(E)` $=$ `numBits(E')`, shape(`operand`) $=$ shape(`result`).
+    * If `numBits(E)` $\gt$ `numBits(E')`, dim(`result`, `i`) $=$
+      dim(`operand`, `i`) for all `i` $\in$ [0, `R`-1), and
+      `dim(result, -1) = numBits(E)/numBits(E')`.
+    * If `numBits(E')` $\gt$ `numBits(E)`, dim(`result`, `i`) $=$
+      dim(`operand`, `i`) for all `i` $\in$ [0, `R`-1), and
+      `dim(operand, -1) = numBits(E')/numBits(E)`.
```

Comment:
Proposed rewording:

```
  * (C1) Let `E` and `E'` be the `operand` and `result` element type,
    respectively and `R = rank(operand)`:
    * If `numBits(E')` $=$ `numBits(E)`, shape(`result`) $=$ shape(`operand`).
    * If `numBits(E')` $\lt$ `numBits(E)`:
      * `rank(result) = R+1`.
      * dim(`result`, `i`) $=$ dim(`operand`, `i`) for all `i` $\in$ [0, `R`-1].
      * `dim(result, R) = numBits(E)/numBits(E')`.
    * If `numBits(E')` $\gt$ `numBits(E)`:
      * `rank(result) = R-1`.
      * dim(`result`, `i`) $=$ dim(`operand`, `i`) for all `i` $\in$ [0, `R`-1).
      * `dim(operand, R-1) = numBits(E')/numBits(E)`.
```

Feedback:
  1) Dropped "`operand` and `result` have the same shape except for the last dimension" because sometimes they have exactly the same shape.
  2) Same as in the formalism, reordered formulae so that result is consistently on the lhs and operand is consistently on the rhs.
  3) Introduced another level of bullets to make it individual cases stand out among prose with ifs and foralls.
  4) Used absolute axes everywhere, i.e. `R-1` instead of `-1`, again for consistency reasons. This uncovered a bug in the second bullet.